### PR TITLE
docs: Add a section for breaking changes in RFC

### PR DESCRIPTION
**Pull request by dequejosie**
Closes: https://github.com/dequelabs/cauldron/issues/797

This is just a draft. Please feel free to comment.

### CODE REVIEW

File: rfcs.md [hunk 1]
```diff
@@ -33,3 +33,55 @@ Some changes that would not need an RFC:
...
+
+1. we might break our API too frequently and that would decrease Cauldron's stability and risk fatiguing our users
+
+2. we assume that the majority of Cauldron users do not consume CSS styles separately from their React components (although we allow our users to do so)
+
+3. we have not been marking the changes to remove or rename a classname as breaking
+
+The counterpoint to the first concern is that major version numbers are not sacred, as the creator of `SemVer` spec [defends releasing a new major version only about one month after the previous one](https://tom.preston-werner.com/2022/05/23/major-version-numbers-are-not-sacred.html). The proponents of `SemVer` spec also point out that we do not have any data to show how exactly our users utilize Cauldron. Therefore the first assumption could be right, but we have no proof for it and we have no control over how ours users consume . The last point is dictated by how we agree on the first two.
```
Comment (isner) [inline rfcs.md:49]: > ...no control over how ours users consume .

Typo?

[the remark this answers is not in the document]
Comment (dequejosie) [reply rfcs.md]: Maybe I didn't represent that side properly. They were mostly referring to the changes that do not have a visual impact. For the cases that triggered the debate, please check https://github.com/dequelabs/cauldron/pull/793 and https://github.com/dequelabs/cauldron/pull/782

[the remark this answers is not in the document]
Comment (isner) [reply rfcs.md]: Thanks - I see. Yeah, that's tricky. Even those minor changes _can_ have a visual impact depending on whether a library consumer is using them directly. This speaks to your point about not knowing how users actually use the library, and to what degree users typically customize the components.

[the remark this answers is not in the document]
Comment (dequejosie) [reply rfcs.md]: Exactly!

File: rfcs.md [hunk 2]
```diff
@@ -33,3 +33,55 @@ Some changes that would not need an RFC:
 The goal of our RFC process is to reach a consensus among the cauldron team on proposed changes, and to ensure that new features are congruent with the rest of Cauldron. The Cauldron team will have a recurring office hours meeting on the 1st Wednesday of each month, giving the team the opportunity to discuss any new RFCs that have been introduced and make comments.
 
 Once a consensus has been reached for the proposed RFC, the `rfc` label will be removed from the issue and the proposed changes can then be implemented.
+
+## What is a Breaking Change?
```
Comment (thuey) [inline rfcs.md:37]: @dequejosie Thanks for putting this together. However, this document is just meant to document our RFC process--not actually contain the RFC's. To submit an RFC, you should follow the steps listed out [above](https://github.com/dequelabs/cauldron/blob/develop/rfcs.md#how-to-submit-an-rfc):

1. Create a new issue
1. Add the `rfc` label
1. Be sure to include descriptive details, this includes:
   - A descriptive title
   - Implementation details
   - Screenshots (if necessary for a new component)

**Comment by dequejosie** [comment]
Submitted a rfc issue. Thus closing this one.

**Pull request closed without merging**